### ISSUE TITLE
Feature: Diagnostics and Metrics view EPIC

### ISSUE THREAD

**Issue by SgtPooki** [ipfs/ipfs-webui#1925]
**Is your feature request related to a problem? Please describe.**
Creating a diagnostic & metrics view in the webui and desktop app

**Describe the solution you'd like**
It's being discussed in https://github.com/ipfs/ipfs-webui/issues/1220

**Describe alternatives you've considered**
* https://pl-diagnose.on.fleek.co/
* https://ipfs-check.on.fleek.co/

**Additional context**

Ideas for supported functionality:

* network connection view to support users determining why or whether they're experiencing network connectivity issues. 
  * see https://github.com/ipfs/ipfs-desktop/search?q=context+deadline+exceeded&type=issues

**Comment by SgtPooki** [comment]
@lidel @BigLep what do you think about prompting current IPFS desktop & webui issues to respond on this and the metrics (#1220) view issue?

@lidel do we have any current methods to prompt users for one-time feedback? i.e. surveys, etc..?

**Comment by SgtPooki** [comment]
Thanks for the great insight! I think adding pl-diagnose into the webui is a great idea. If [pl-diagnose](https://github.com/laurentsenta/pl-diagnose)'s components were abstracted, we could both benefit from the work already done on pl-diagnose.

@laurentsenta are you still actively working on that project? How hard do you think it would be to pull out some components from pl-diagnose so we could re-use them in ipfs-webui? 

@lidel @laurentsenta would it make sense to fully consume pl-diagnose (the https://pl-diagnose.on.fleek.co/#/diagnose/access-content page) as a screen in the webui and ipfs-desktop? If there are still plans to expand pl-diagnose, maybe it doesn't make sense, but hosting pl-diagnose within the webui, desktop app, and at ipns://webui.ipfs.io/#/diagnose seems like a great idea to me. 

I feel like it would be a natural progression for the webui and desktop diagnostics screen to expand beyond simply troubleshooting content accessibility, but I would love to hear what the future for pl-diagnose is.

**Comment by BigLep** [comment]
Cc @aschmahmann since would ideally like to see convergence around one main IPFS diagnostics tool in EngRes. 
 I believe https://github.com/ipfs-shipyard/ipfs-check has been the one to get the most investment.

**Comment by aschmahmann** [comment]
Agreed, from a practical perspective I'd rather only have to maintain one tool.

Happy to take PRs to ipfs-check. Will also allow some feedback from the other folks who have contributed to ipfs-check

**Comment by SgtPooki** [comment]
@laurentsenta it would be great if we could simply import pl-diagnose as a full-page view. This is probably the quickest and easiest option.

Moving forward, having that view be separated into different imported components would allow us to control the flow better as our diagnostic needs change .

It sounds like you've already got four react components, so releasing those as a library to npm should probably be our first step.

**Comment by juliaxbow** [comment]
re: user research around webui/desktop diagnostics view

I'm still hoping to get more [survey](https://cryl27xbs16.typeform.com/to/VFVQeYqe?typeform-source=admin.typeform.com) responses as well as source additional participants for addt'l user testing but will deliver some research findings next week, fyi

**Comment by SgtPooki** [comment]
> Cc @aschmahmann since would ideally like to see convergence around one main IPFS diagnostics tool in EngRes. I believe https://github.com/ipfs-shipyard/ipfs-check has been the one to get the most investment.

> Agreed, from a practical perspective I'd rather only have to maintain one tool.
> 
> Happy to take PRs to ipfs-check. Will also allow some feedback from the other folks who have contributed to ipfs-check

@BigLep @tinytb @lidel 

I agree that maintaining a single tool is ideal. The problem is where that tool exists in the stack and what diagnostics tooling is needed specifically within IPFS desktop and webui because they may not make sense elsewhere:

1. CLI/REPL inside desktop/webui to assist with learning/debugging/other
2. Network troubleshooting of the kubo daemon controlled by ipfs desktop
3. Diagnostics tooling for troubleshooting of MFS files added via IPFS Desktop
4. Other issues we're having trouble gathering (maybe they don't exist?)

To get our "one tool" we need to consolidate ipfs-check, pl-diagnose, and the needs of webui+desktop. If ipfs-check is the best place for that, great. I don't see that being our(webui & ipfs in general) path to success, but I'm open to being convinced.

If we need to separate the discussion of general ipfs diagnostics tooling, and this issue -- which I intended to be specific to the tooling that would be useful in IPFS Desktop -- then let's do that. If we've reached the end of the discussion because there aren't any diagnostics tooling that make the most sense inside of IPFS desktop and webui, then I'm good with closing this issue and pushing on consolidation of diagnostics tooling inside of ipfs-check.

**Comment by SgtPooki** [comment]
Related: @juliaxbow survey and user research:
![image](https://user-images.githubusercontent.com/1173416/195155641-a5b6701b-c1f4-4a34-9191-e1d0ca6c8e29.png)

The users we heard back from seem to want diagnostics inside desktop/webui. So are the following next steps valid?

1. Consolidate diagnostics in ipfs-check 
2. Render ipfs-check inside webui&desktop

Or should we skip the second item altogether?

**Comment by 2color** [comment]
> Consolidate diagnostics in ipfs-check

Yes! I think we all agree on this one.

> Render ipfs-check inside webui&desktop

I think it would be useful to have something inside the webui. The challenge is that ipfs-check relies on its own backend so if you render it do you use the centralised backend? Because technically if you have access to the webui, you probably have access to Kubo's RPC API which you can use to debug too.

**Comment by SgtPooki** [comment]
FYI to all following this thread that this "diagnostics view" may well turn more into an effort of building some consolidated diagnostics tooling rather than a specific "diagnostics view" within webui. I feel like both efforts have merit, but from discussions during [roadmap planning](https://github.com/ipfs/ipfs-gui/issues/102) that seems like where we're going

p.s. discovered a tool that I hadn't seen before, https://dag.ipfs.tech/, which has source at https://github.com/ipfs-shipyard/ipfs-dag-builder-vis, that would make sense in either effort

**Comment by SgtPooki** [comment]
@lidel We could revive this issue for integrating check.ipfs.network into ipfs-webui view

**Comment by SgtPooki** [comment]
FYI that this is related to https://github.com/ipshipyard/roadmaps/issues/11 and has significantly reduced in scope, however, we will start off with a live log trailing viewer that I have working locally and am working on getting a PR out for:

https://github.com/user-attachments/assets/4443feaf-6bc9-48b1-9c92-efdefc1d3270
